CMakeLists.txt

```diff
@@ -238,16 +238,19 @@ target_compile_definitions(devilution PUBLIC
 target_compile_definitions(devilutionx PRIVATE ASIO_STANDALONE)
 
 
-if(USE_SDL1)
-  target_link_libraries(devilutionx PRIVATE
-    ${SDL_LIBRARY} ${SDL_TTF_LIBRARY} ${SDL_MIXER_LIBRARY})
-  target_compile_definitions(devilutionx PRIVATE USE_SDL1)
-else()
-  target_link_libraries(devilutionx PRIVATE
-    SDL2::SDL2main
-    SDL2::SDL2_ttf
-    SDL2::SDL2_mixer)
-endif()
+foreach(target devilution devilutionx)
+  if(USE_SDL1)
+    target_link_libraries(${target} PRIVATE
+      ${SDL_LIBRARY} ${SDL_TTF_LIBRARY} ${SDL_MIXER_LIBRARY})
+    target_compile_definitions(devilutionx PRIVATE USE_SDL1)

```

Comment:
This line is duplicate

Reply:
Fixed, also didn't have the variable and the backports header wasn't included

Reply:
Nice now there is just the issue witht Mac build left,  I asked ErgoDon about it previously and he had this to say:

> It's a weird behaviour but apparently in devilution lib the system has not identify the compiler, so when we include sdl.h, internally, sdl needs include a header with Apple macros and this header is the problem, if we force the macro "_ GNUC _" the project compile. I don´t found the reason of this 'cause has the same behaviour when build the x86 version from Xcode or with the xcode-build tool
The first approach can be do this temporally until find  some cause oft his https://github.com/BennyFranco/devilutionX/commit/eb4831f25f29d0f9a23e4ed61c6b9360544f2962?diff=unified#diff-af3b638bc2a3e6c650974192a53c7291R290
(sorry for the unnecessary tabulator, my editor betrayed me)
Apparently works https://travis-ci.com/BennyFranco/devilutionX/builds/127660501

Reply:
Isn't this because we're passing `-fms-compatibility`?

Reply:
I don't know enough about the build system to answer that. A few flags can definitely be removed at this point since the code is in a MUCH better state since the build system was originally setup.

Reply:
OK so on travis build we're using a compiler from XCode with name "gcc" but actually CMake says:

-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046

Hmm

Reply:
Yeah, Apple made de switch a few years ago, they like to do hard breaks so the only level of compatibility they provide is mislabeling things. I'll test a bit to see what can be changed about the build flags.

Reply:
I've managed to remove the MS compat flags in the latest commit, hope it works.

Reply:
It works!